ndcube/global_coords/global_coords.py

```diff
@@ -0,0 +1,74 @@
+from collections import OrderedDict
+from collections.abc import Mapping
+
+__all__ = ['GlobalCoords']
+
+
+class GlobalCoords(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+    """
+    def __init__(self, ndcube):
+        """
+        Init method.
+        """
+        super().__init__()
+        self._ndcube = ndcube
+
+        # Set values using the setters for validation
+        self._all_coords = OrderedDict()
```

Comment:
This and the usage in `add` and `remove` as I mentioned before should still be `_internal_coords`. The idea here is that GlobalCoords has multiple sources of data. The one being implemented here is the internal list of coords which users can manually add and remove from. In follow up PRs we will add dynamically getting coords from the wcs and extra_coords of the ndcube.

So we want a property which collates all these sources of coords into one `_all_coords` and one attribute which stores the internal state the user can add and remove from. *Therefore* the `add` and `remove` methods mutate the internal state but all the other methods use the combined set of coords returned by the `_all_coords` property.

Reply:
Ah, I see. Let me make the changes now...

Reply:
Fixed... If I have omitted anything, please let me know.